mkosi

```diff
@@ -1258,24 +1258,28 @@ def install_arch(args, workspace, run_build_script):
     if args.release is not None:
         sys.stderr.write("Distribution release specification is not supported for Arch Linux, ignoring.\n")
 
-    keyring = "archlinux"
-
-    if platform.machine() == "aarch64":
-        keyring += "arm"
-
-    run(["pacman-key", "--nocolor", "--init"], check=True)
-    run(["pacman-key", "--nocolor", "--populate", keyring], check=True)
-
     if platform.machine() == "aarch64":
         server = "Server = {}/$arch/$repo".format(args.mirror)
     else:
         server = "Server = {}/$repo/os/$arch".format(args.mirror)
 
-    with open(os.path.join(workspace, "pacman.conf"), "w") as f:
+    root = os.path.join(workspace, "root")
+    # Create base layout for pacman-key
+    os.makedirs(os.path.join(root, "var/lib/pacman"), 0o755, exist_ok=True)
+    os.makedirs(os.path.join(root, "etc/pacman.d/gnupg"), 0o755, exist_ok=True)
+    # Fix permissions when subvolumes are used
+    os.chmod(os.path.join(root, "var/lib"), 0o755)
+    os.chmod(os.path.join(root, "var/tmp"), 0o1777)
```

Comment:
this breaks when output format is not a subvolume. Is this really fixing something? In prepare_tree() this is already the permission used.

Reply:
I'll look into it.

The `chmod`s are just there to fix a few warnings during the installation and could be dropped or better handled.

The makedirs however is necessary to set up the keyring inside the image.